magicctapipe/scripts/lst1_magic/semi_automatic_scripts/DL1_to_DL2.py

```diff
@@ -0,0 +1,302 @@
+"""
+This script creates the bashscripts necessary to apply "lst1_magic_dl1_stereo_to_dl2.py"
+to the DL1 stereo data. It also creates new subdirectories associated with
+the data level 2.
+
+Usage:
+$ DL1_to_DL2 -c configuration_file.yaml
+"""
+import argparse
+import datetime
+import glob
+import logging
+import os
+from pathlib import Path
+
+import joblib
+import numpy as np
+import pandas as pd
+import yaml
+
+from magicctapipe import __version__
+from magicctapipe.io import resource_file
+from magicctapipe.scripts.lst1_magic.semi_automatic_scripts.clusters import (
+    rc_lines,
+    slurm_lines,
+)
+
+__all__ = ["ST_NSB_List", "bash_DL1Stereo_to_DL2"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def ST_NSB_List(
+    target_dir, nsb_list, nsb_limit, source, df_LST, ST_list, ST_begin, ST_end, version
+):
+    """
+    This function creates the lists of runs separeted by run period and NSB level.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    nsb_list : list
+        List of the MC NSB values
+    nsb_limit : list
+        Edges of the NSB binning
+    source : str
+        Source name
+    df_LST : :class:`pandas.DataFrame`
+        Dataframe collecting the LST1 runs (produced by the create_LST_table script)
+    ST_list : list
+        List of the observastion periods
+    ST_begin : list
+        List of beginning dates for the observation periods
+    ST_end : list
+        List of ending dates for the observation periods
+    version : str
+        Version of the input (stereo subruns) data
+    """
+
+    # Loops over all runs of all nights
+    Nights_list = np.sort(
+        glob.glob(f"{target_dir}/v{version}/{source}/DL1Stereo/Merged/*")
+    )
+    for night in Nights_list:
+        # Night period
+
+        night_date = night.split("/")[-1]
+        outdir = f"{target_dir}/v{__version__}/{source}/DL2/{night_date}/logs"
+        os.makedirs(outdir, exist_ok=True)
+        date_lst = night_date[:4] + "_" + night_date[4:6] + "_" + night_date[6:8]
```

Comment:
you use this date_lst only to convert this to datetime object, so I think you can strip time directly from night_date:
```
date_magic = datetime.datetime.strptime(
            night_date, "%Y%m%d"
        ) + datetime.timedelta(days=1)
```